### DISCUSSION TITLE
Get Student ID from submission metadata

### DISCUSSION THREAD

**Discussion by emilioapontea** [gradescope/autograder_samples#118] [Q&A]
I am working on an autograder inherited from a previous semester of a course. This autograder script worked in the past, but we are running into some issues with obtaining the submission metadata through the `autograder/submission_metadata.json` file. I have confirmed that most of the metadata information is correct, except for the `metadata['users'][0]['sid']` value. This is an individual assignment so the `metadata['users']` list contains a single student dictionary, although I imagine the error is not related to the number of students in the list. Even though I cannot find a reference to this dictionary key in the [documentation](https://gradescope-autograders.readthedocs.io/en/latest/submission_metadata/), I have confirmed that the student dictionary does contain the `['sid']` key, but is always paired with the value `None`. The autograder is supposed to obtain the student's campus ID from the metadata, and the Gradescope course roster is configured with student IDs. Is there a way to obtain the student's student ID from the submission metadata file?

**Comment by mattrussell2** [comment]
From the documentation, the `users` field of the `metadata.json` file looks like this:

```{json}
"users": [
    {
      "email": "student@example.com",
      "id": 1234,
      "name": "Student User"
    }, ... // Multiple users will be listed in the case of group submissions
  ],
```

There is no `sid` field, however there is an `id` field. Assuming that your students were working individually, then you will be able to access

```
metadata['users'][0]['id']     # student's id
metadata['users'][0]['name']   # student's name
metadata['users'][0]['email']  # student's email
```

**Comment by adam-brucker** [comment]
Do you use an LMS?  I've seen it in the past where Canvas does not send that value over.
